tsdb/record/record_test.go

```diff
@@ -160,3 +184,71 @@ func TestRecord_Type(t *testing.T) {
 	recordType = dec.Type([]byte{0})
 	require.Equal(t, Unknown, recordType)
 }
+
+func TestRecord_MetadataDecodeUnknownFields(t *testing.T) {
```

Comment:
This unit test is a bit fragile because of manual encoder writes. It would be better to use the `encoder.Metadata` function to encode the metadatas instead of manually writing the records. Any reason why it is like this?

Reply:
This is tied to my above comment. We've used the encoder.Metadata method to directly test the encoding/decoding of metadata in `TestRecord_EncodeDecode`. Here we're manually writing the records so that we can assert that unknown fields will be skipped correctly. Does it make any sense?